neural_compressor/torch/quantization/quantize.py

```diff
@@ -59,11 +61,19 @@ def quantize(
     logger.info(quant_config.to_dict())
     # select quantization algo according to config
 
-    model_info = quant_config.get_model_info(model=q_model)
+    if is_ipex_available:
+        model_info = quant_config.get_model_info(q_model, example_inputs)
+    else:
+        model_info = quant_config.get_model_info(model=q_model)
     configs_mapping = quant_config.to_config_mapping(model_info=model_info)
     logger.debug(configs_mapping)
     for algo_name, algo_func in algos_mapping.items():
         if need_apply(configs_mapping, algo_name):
             logger.info(f"Start to apply {algo_name} on the model.")
-            q_model = algo_func(q_model, configs_mapping, run_fn=run_fn, run_args=run_args)
+            if is_ipex_available:
+                q_model = algo_func(
+                    q_model, configs_mapping, run_fn=run_fn, run_args=run_args, example_inputs=example_inputs
+                )
+            else:
+                q_model = algo_func(q_model, configs_mapping, run_fn=run_fn, run_args=run_args)
```

Comment:
We'd better unify the `algo_func` APIs, each `algo_func` within the `quantization/algorithm_entry.py` passes the necessary args to its corresponding implementation.


Something like below:
```python
# algorithm_entry.py
@register_algo(STATIC_QUANT)
def gptq_entry(
    model: torch.nn.Module,
    configs_mapping: Dict[Tuple[str, callable], StaticQuantConfig],
    run_fn=None,
    run_args=None,
    example_inputs=None,
    inplace=True,
) -> torch.nn.Module:
    ...
    # not pass `examples_inputs`
    gptq_impl(model, configs_mapping, run_fn, run_args)

@register_algo(STATIC_QUANT)
def static_quant_entry(
    model: torch.nn.Module,
    configs_mapping: Dict[Tuple[str, callable], StaticQuantConfig],
    run_fn=None,
    run_args=None,
    example_inputs=None,
    inplace=True,
) -> torch.nn.Module:
    ...
    # pass `examples_inputs`
    static_quant_impl(model, configs_mapping, run_fn, run_args, example_inputs, inplace)

```